python/pyspark/mllib/linalg/__init__.py

```diff
@@ -404,11 +414,57 @@ def __repr__(self):
         return "DenseVector([%s])" % (', '.join(_format_float(i) for i in self.array))
 
     def __eq__(self, other):
-        return isinstance(other, DenseVector) and np.array_equal(self.array, other.array)
+        """
+        Test DenseVector for equality.
+
+        >>> v1 = DenseVector([0.0, 1.0, 0.0, 5.5])
```

Comment:
Please move the test to `tests.py` because they doesn't need to be part of the API doc.